elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/ElideAsyncConfiguration.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.spring.config;
+
+import com.yahoo.elide.Elide;
+import com.yahoo.elide.async.models.AsyncQuery;
+import com.yahoo.elide.async.models.AsyncQueryResult;
+import com.yahoo.elide.async.service.AsyncExecutorService;
+import com.yahoo.elide.core.EntityDictionary;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Async Configuration For Elide Services.  Override any of the beans (by defining your own) to change
+ * the default behavior.
+ */
+@Configuration
+@EntityScan(basePackageClasses = AsyncQuery.class)
+@EnableConfigurationProperties(ElideConfigProperties.class)
+@ConditionalOnExpression("${elide.async.enabled:false}")
+public class ElideAsyncConfiguration {
+
+    /**
+     * Updates the entity dictionary for Elide by binding the Async Models.
+     * @param dictionary Contains the static metadata about Elide models.
+     * @return updated instance of dictionary.
+     */
+    public EntityDictionary bindAsyncDictionary(EntityDictionary dictionary) {
+
+        dictionary.bindEntity(AsyncQuery.class);
```

Comment:
If we are using the JpaDataStore, these classes are automatically discovered.  There is no need to explicitly bind them.

Reply:
We should be able to remove this function

Reply:
Did this because we did not want the Async Models to be discovered/binded if the enableAsync is set to false.

Reply:
@aklish By default Spring scans for entities in the package where App.java (with annotation `@SpringBootApplication`) will be placed.  so it will not be scanning and adding these entities automatically.